docs/understand/texture_fetching.rst

```diff
@@ -0,0 +1,107 @@
+.. meta::
+  :description: This chapter describes the texture fetching modes of the HIP ecosystem
+  ROCm software.
+  :keywords: AMD, ROCm, HIP, Texture, Texture Fetching
+
+*******************************************************************************
+Texture Fetching
+*******************************************************************************
+
+`Textures <https://rocm.docs.amd.com/projects/HIP/en/latest/doxygen/html/group___texture.html>`_ are more than just a buffer, that is interpreted as 1D, 2D or 3D array. Because of its legacy as a graphics functionality, textures are indexed with floats. This can happen two different ways, either the index will be in the range of [0..size-1] or in [0..1]. The difference is mathematically just a division, so for the explanations on this page, we'll use the [0..size-1] indices.
+
+Using float indices isn't trivial. The two issues that come up are sampling and addressing.
+
+When a texture is indexed with a fraction, the queried value is often between two or more texels (texture elements). The sampling method defines what value to return in these cases.
+
+Sometimes the index is outside the bounds of the texture. This might seem unintuitive, but is useful in computer graphics for putting a texture on a surface multiple times, or just creating a visible sign of out of bounds indexing. Addressing mode defines what value to return when indexing a texture out of bounds.
+
+There are several different sampling and addressing modes, and no rule for how to choose between them, as it is dependent on the application. This means that it has to be decided during development which sampling and addressing method to use.
+
+This image is the texture, that will be used as for the examples below. It is 2x2 texels and is indexed in the [0..1] range.
+
+.. image:: ../data/understand/textures/original.png
+  :width: 150
+  :alt: Texture used as example
```

Comment:
An additional caption for these pictures would be nice. Basically the same as the alt-text already states